File: ios/Sources/Model/JSOpenId4VpConfig.swift
struct JSOpenId4VpConfig {
    let scheme: String?
    let encryptionAlgorithms: [String]?
    let encryptionMethods: [String]?
    let clientIdSchemes: [JSClientIdScheme]?
}

extension JSOpenId4VpConfig {
    init(fromJson json: JSONDictionary) throws {
        let clientIdSchemesJson = json["clientIdSchemes"] as? [JSONDictionary]
        let clientIdSchemes = try clientIdSchemesJson?.map { try JSClientIdScheme(fromJson: $0) }
        
        self.init(
            scheme: json["scheme"] as? String,
            encryptionAlgorithms: json["encryptionAlgorithms"] as? [String],
            encryptionMethods: json["encryptionMethods"] as? [String],
            clientIdSchemes: clientIdSchemes
        )
    }
}

File: android/src/main/java/com/dsr/eudi/walletkit/reactnative/model/JSOpenId4VpConfig.kt
package com.dsr.eudi.walletkit.reactnative.model

import com.dsr.eudi.walletkit.reactnative.utils.parseEncryptionAlgorithm
import com.dsr.eudi.walletkit.reactnative.utils.parseEncryptionMethod
import eu.europa.ec.eudi.wallet.transfer.openid4vp.OpenId4VpConfig

data class JSOpenId4VpConfig(
  val scheme: String?,
  val encryptionAlgorithms: List<String>,
  val encryptionMethods: List<String>,
  val clientIdSchemes: List<JSClientIdScheme>,
) {
  fun toOpenId4VpConfig(): OpenId4VpConfig {
    var configBuilder = OpenId4VpConfig.Builder()
      .withEncryptionAlgorithms(encryptionAlgorithms.map { parseEncryptionAlgorithm(it) })
      .withEncryptionMethods(encryptionMethods.map{ parseEncryptionMethod(it) })
      .withClientIdSchemes(clientIdSchemes.map { it.toClientIdScheme() })

    if (scheme !== null) {
      configBuilder = configBuilder.withScheme(scheme)
    }

    return configBuilder.build()
  }
}
